eft/preprocess.py

```diff
@@ -52,33 +75,41 @@ def main(args) -> None:
             'start': row.start,
             'end': row.end,
             'values': binned_values
-            }
-        seq_values.append(row_data)
-
-    seq_values = pd.DataFrame(seq_values)
-    seq_values.to_pickle(args.outdir / f"{i}.pkl")
-    seq_values[['chrom', 'start', 'end', 'values']].to_csv(args.outdir / f"{i}.bed", sep="\t", header=False, index=False)
-
+        }
+        promoters.append(row_data)
 
+    len_seq_values = len(promoters)
 
-if __name__ == "__main__":
-    """
-    Parse Arguments
-    
-    """
-    desc = "Preprocess data."
-    parser = argparse.ArgumentParser(
-        description=desc, formatter_class=argparse.ArgumentDefaultsHelpFormatter
-    )
-    
-    parser.add_argument("--outdir", type=Path, help="Directory to write data.")
-    args = parser.parse_args()
-    sys.exit(main(args))
-
+    chrom_sizes = get_chrom_sizes()
 
+    control = []
+    for i in tqdm(range(len_seq_values)):
+        find_control = True
+        while find_control:
+            chrom, start, end, values = random_region(chrom_sizes, bw_file)
 
+            if np.any(np.isnan(values)):
+                continue
+            binned_values = avg_bin(values, n_bins=896)
 
+            if np.all(np.array(binned_values) == 0):
+                find_control = False
+                row_data = {
+                    'chrom': chrom,
+                    'start': start,
+                    'end': end,
+                    'values': values
+                }
+                control.append(row_data)
 
+    promoters = pd.DataFrame(promoters)
+    control = pd.DataFrame(control)
 
```

Comment:
I think we should label the sequence type and then just concat and shuffle these into a single file. We will be splitting this file into `train` and `val` after loading anyways:
```
promoter['seq_type'] = 'promoter'
control['seq_type'] = 'random'
sequences = pd.concat((promoter, control)).sample(frac=1)
```